src/main/java/animals/TextInterface.java

```diff
@@ -0,0 +1,127 @@
+package animals;
+
+import java.text.MessageFormat;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class TextInterface extends LanguageRules {
```

Comment:
Can you rethink the name for the class, because that's conflicted with Interface meaning in Java?